dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go

```diff
@@ -291,11 +293,20 @@ func generateTarball(hadoopClients []string) error {
 	}
 
 	mvnArgs := getCommonMvnArgs(hadoopVersion)
-	run("compiling repo", "mvn", mvnArgs...)
+	if skipUIFlag {
+		mvnArgsNoUI := append(mvnArgs, "-pl", "!webui")
+		run("compiling repo without UI", "mvn", mvnArgsNoUI...)
+	} else {
+		run("compiling repo", "mvn", mvnArgs...)
+	}
+
 	// Compile ufs modules for the main build
 	buildModules(srcPath, "underfs", "hdfs", ufsModulesFlag, version, ufsModules, mvnArgs)
 
 	tarball := strings.Replace(targetFlag, versionMarker, version, 1)
+	if skipUIFlag {
+		tarball = strings.Replace(tarball, version, version+"-noUI", 1)
```

Comment:
I'm concerned with this replace being potentially ambiguous, so should we do something like:
```golang
versionString := version
if skipUIFlag {
  versionString = versionString + "-noUI"
}
tarball := strings.Replace(targetFlag, versionMarker, versionString, 1)
```

Reply:
done